ui/src/app/views/cohort-edit/component.html

```diff
@@ -1,9 +1,12 @@
 <h2>Set Cohort Info</h2>
-<div *ngIf="user; else loggedout">
 <div class="form-area">
   <div>
-    <label >Cohort Name:</label>
-    <input type="text" id="name" class="input" name="cohort-name" [(ngModel)]="cohort.name" autofocus>
+    <label>Cohort Name:</label>
+    <input type="text" id="name" class="input" name="cohort-name" autofocus
+        [(ngModel)]="cohort.name" (ngModelChange)="autoGenerateId()">
```

Comment:
If we do want to autoGenerate the ID here, there is a boolean variable set on init that's job is to say that we are adding the cohort vs editing an existing cohort. Wrapping the autoGenerateId function in an if statement can ensure the ID is only set once.